source/fundamentals/odata.txt

```diff
@@ -0,0 +1,203 @@
+free MongoDB Atlas cluster and load the sample datasets, see the :ref:`<csharp-quickstart>`.
+
+Tutorial
+--------
+
+.. procedure::
+
+   .. step:: Install Dependencies
+
+      Create a new ASP.Net application named
+      ``ODataExample`` and install the {+driver-short+}. You can install the
+      driver by using the NuGet package manager in your IDE, or by running the
+      following command in the .NET CLI:
+
+      .. code-block:: shell
+    
+         dotnet add package MongoDB.Driver
+      
+      Then, install the ``MongoDB.AspNetCore.OData`` NuGet
+      package through the NuGet Package Manager or through the .NET CLI by running the
+      following command:
+
+      .. code-block:: shell
+
+         dotnet add package MongoDB.AspNetCore.OData
+
+   .. step:: Define your Models
+
+      Create a new folder in your solution called ``Models`` and copy the
+      following ``Restaurant.cs``, ``Address.cs``, and ``GradeEntry.cs`` files into the
+      folder:
+
+      .. literalinclude:: /includes/code-examples/RestaurantOdata.cs
+         :language: csharp
+         :copyable:
+         :dedent:
+
+      .. literalinclude:: /includes/code-examples/Address.cs
+         :language: csharp
+         :copyable:
+         :dedent:
+
+      .. literalinclude:: /includes/code-examples/GradeEntry.cs
+         :language: csharp
+         :copyable:
+         :dedent:
+
+      .. include:: /includes/convention-pack-note.rst
+
+   .. step:: Create an OData Controller
+
+      Create a new folder in your solution called ``Controllers`` and add a new
+      controller file called ``RestaurantsController.cs``. Copy the following code
+      into the file:
+
+      .. literalinclude:: /includes/fundamentals/code-examples/connection/OData.cs
+         :language: csharp
+         :start-after: // start-controller
+         :end-before: // end-controller
+
+      This code performs the following actions:
+
+      - Creates a constructor that connects to MongoDB, and gets the
+        ``restaurants`` collection.
+      - Creates a ``Get`` endpoint that returns all restaurants in the collection.
+      - Specifies the ``MongoEnableQuery`` attribute to enable querying on the
+        ``Get`` endpoint.
+      - Specifies the ``PageSize`` attribute on ``MongoEnableQuery`` to limit the
+        number of documents returned to ``5``.
+
+   .. step:: Configure the OData Service
+
+      Paste the following code into your ``Program.cs`` file to configure the OData
+      service and map your controller endpoints.
+
+      .. literalinclude:: /includes/fundamentals/code-examples/connection/OData.cs
+         :language: csharp
+         :start-after: // start-configure
+         :end-before: // end-configure
+
+      .. note::
+
+         Replace the ``<"Your connection URI">`` placeholder with your MongoDB connection string.
+      
+      This code performs the following actions:
+
+      - Instantiates a new ``MongoClient`` and passes it to the controller.
```

Comment:
Instantiates a new ``MongoClient`` and register it as a singleton in the DI container.